### PR TITLE
refactor: remove smithy-bindgen dependency

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

Switch back to using `wasmcloud-interface-testing` instead of `smithy_bindgen`

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: wasmcloud-test-util/Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,10 @@ repository = "https://github.com/wasmcloud/wasmcloud-test"
 readme = "README.md"
 
 [dependencies]
-smithy-bindgen = "0.1.0"
 serde_bytes = "0.11"
 regex = "1"
 wasmbus-rpc = { version = "0.14", features = ["otel"]}
+wasmcloud-interface-testing = "0.8.0"
```
Comment (connorsmith256) [inline wasmcloud-test-util/Cargo.toml:16]: This uses wasmbus-rpc 0.13, doesn't it? I'm surprised there wasn't a conflict here, and are we sure it doesn't create one downstream?

- [ ] double check consumers of this crate can rely on wasmbus-rpc 0.14 and this branch of wasmcloud-test-util

File: wasmcloud-test-util/src/lib.rs [hunk 2]
```diff
@@ -8,66 +8,8 @@ pub mod cli;
 pub use nkeys;
 pub use regex;
 
-pub mod testing {
-
-    smithy_bindgen::smithy_bindgen!("testing/testing.smithy", "org.wasmcloud.interface.testing");
-
-    // after sdk is split we won't have to duplicate this code
-    impl Default for TestOptions {
```
Comment (connorsmith256) [inline wasmcloud-test-util/src/lib.rs:16]: I think losing these will break downstream consumers, e.g. https://github.com/wasmCloud/capability-providers/blob/0144fea03d2eebfe8b3c91bd381b400a8ce2479a/nats/tests/nats_test.rs/#L14

**Review by connorsmith256 — CHANGES_REQUESTED** [review]

File: wasmcloud-test-util/Cargo.toml [hunk 3]
```diff
@@ -10,10 +10,10 @@ repository = "https://github.com/wasmcloud/wasmcloud-test"
 readme = "README.md"
```
Comment (connorsmith256) [inline wasmcloud-test-util/Cargo.toml:10]: we'll need to bump the version to 0.10.0 before releasing

File: wasmcloud-test-util/src/lib.rs (hunk 2 above)
Comment (vados-cosmonic) [reply wasmcloud-test-util/src/lib.rs:16]: I'll roll this into the downstream crates check

File: wasmcloud-test-util/Cargo.toml (hunk 1 above)
Comment (vados-cosmonic) [reply wasmcloud-test-util/Cargo.toml:16]: Looks like there are two public consumers of `wasmcloud-interface-testing`, `wash-cli` and `wasmcloud-provider-httpserver` -- right now `wasmcloud-provider-httpserver` also uses `wasmbus-rpc` version `0.13` -- you're right that there is at least one conflict though, but it has to do with `async-nats`:

```
error[E0599]: no method named `with_client_auth_cert` found for struct `ConfigBuilder` in the current scope
   --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.30.0/src/tls.rs:117:29
    |
117 |                     builder.with_client_auth_cert(cert, key).map_err(|_| {
    |                             ^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `with_no_client_auth`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `async-nats` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
```

Also, turns out [`wasmcloud-test-util` is the only thing that depends on `wasmbus` `0.14` right now](https://crates.io/crates/wasmbus-rpc/reverse_dependencies).

File: wasmcloud-test-util/Cargo.toml (hunk 3 above)
Comment (vados-cosmonic) [reply wasmcloud-test-util/Cargo.toml:10]: Sure going to put it in here just to make things faster.

File: wasmcloud-test-util/src/lib.rs (hunk 2 above)
Comment (vados-cosmonic) [reply wasmcloud-test-util/src/lib.rs:16]: Yup this does break the downstream consumers -- They all have to be upgraded together.

Getting the NATS provider to build required the following changes:

in `wasmcloud-provider-nats`:
- `async-nats=0.30`
- `wasmbus-rpc=0.14`
- `wasmcloud-interface-messaging=custom`
- `wasmcloud-test-util=custom`

in `wasmcloud-interface-messaging=custom`:
- `wasmbus-rpc=0.14`

in `wasmcloud-test-util`:
- `wasmcloud-interface-testing=custom`
- `wasmbus-rpc=0.14`

in `wasmcloud-interface-testing`:
- `wasmbus-rpc=0.14`

IIRC there was going to be push to bump all of these.

Long term I think we need to remove some of the macros and start using some traits mroe,

**Pull request merged**